Core.ApplicationServices/Organizations/Write/OrganizationWriteService.cs

```diff
@@ -0,0 +1,380 @@
+﻿using System;
+using Core.Abstractions.Types;
+using Core.ApplicationServices.Authorization;
+using Core.ApplicationServices.Extensions;
+using Core.ApplicationServices.Model.Organizations;
+using Core.ApplicationServices.Model.Organizations.Write;
+using Core.ApplicationServices.Model.Organizations.Write.MasterDataRoles;
+using Core.DomainModel;
+using Core.DomainModel.Events;
+using Core.DomainModel.Organization;
+using Core.DomainServices;
+using Core.DomainServices.Generic;
+using Core.DomainServices.Repositories.Organization;
+using Infrastructure.Services.DataAccess;
+
+namespace Core.ApplicationServices.Organizations.Write;
+
+public class OrganizationWriteService : IOrganizationWriteService{
+        
+    private readonly ITransactionManager _transactionManager;               
+    private readonly IDomainEvents _domainEvents;
+    private readonly IOrganizationService _organizationService;
+    private readonly IAuthorizationContext _authorizationContext;
+    private readonly IOrganizationRepository _repository;
+    private readonly IEntityIdentityResolver _identityResolver;
+    private readonly IGenericRepository<ContactPerson> _contactPersonRepository;
+    private readonly IGenericRepository<DataResponsible> _dataResponsibleRepository;
+    private readonly IGenericRepository<DataProtectionAdvisor> _dataProtectionAdvisorRepository;
+
+
+    public OrganizationWriteService(ITransactionManager transactionManager, IDomainEvents domainEvents, IOrganizationService organizationService, IAuthorizationContext authorizationContext, IOrganizationRepository repository, IEntityIdentityResolver identityResolver, IGenericRepository<ContactPerson> contactPersonRepository, IGenericRepository<DataResponsible> dataResponsibleRepository, IGenericRepository<DataProtectionAdvisor> dataProtectionAdvisorRepository)
+    {
+        _transactionManager = transactionManager;
+        _domainEvents = domainEvents;
+        _organizationService = organizationService;
+        _authorizationContext = authorizationContext;
+        _repository = repository;
+        _identityResolver = identityResolver;
+        _contactPersonRepository = contactPersonRepository;
+        _dataResponsibleRepository = dataResponsibleRepository;
+        _dataProtectionAdvisorRepository = dataProtectionAdvisorRepository;
+    }
+
+    public Result<Organization, OperationError> UpdateMasterData(Guid organizationUuid, OrganizationMasterDataUpdateParameters parameters)
+    {
+        using var transaction = _transactionManager.Begin();
+        var result = _organizationService.GetOrganization(organizationUuid)
+            .Bind(organization => Update(organization, parameters));
+
+        if (result.Ok) transaction.Commit();
```

Comment:
Maybe do a if(result.Failed) {rollback; return error;}

transaction.Commit();
return result;

I feel like it's more consistent (I'm just not a fan of if else :D)